src/highlight.rs

```diff
@@ -14,42 +15,65 @@ use std::path::Path;
 use std::rc::Rc;
 use unicode_segmentation::UnicodeSegmentation;
 
-type LineHighlights<'a> = (&'a str, Vec<(Range<usize>, Style)>);
-
-pub(crate) fn highlight_hunk_lines<'a>(
-    config: &'a Config,
-    diff: &'a Rc<Diff>,
-    file_i: usize,
-    hunk_i: usize,
-) -> impl Iterator<Item = LineHighlights<'a>> + 'a {
-    let old_path = &diff.text[diff.file_diffs[file_i].header.old_file.clone()];
-    let new_path = &diff.text[diff.file_diffs[file_i].header.new_file.clone()];
-
-    let hunk = &diff.file_diffs[file_i].hunks[hunk_i];
-    let hunk_content = &diff.text[hunk.content.range.clone()];
-    let old_mask = diff.mask_old_hunk(file_i, hunk_i);
-    let new_mask = diff.mask_new_hunk(file_i, hunk_i);
+#[cached(
+    ty = "SizedCache<String, HunkHighlights>",
+    create = "{ SizedCache::with_size(200) }",
+    convert = r#"{ format!("{_hunk_hash}") }"#
```

Comment:
```suggestion
    ty = "SizedCache<u64, HunkHighlights>",
    create = "{ SizedCache::with_size(200) }",
    convert = r#"{ _hunk_hash }"#
```

I suspect something like this might work now

Reply:
You're correct! I completely missed this :sweat_smile: Done in https://github.com/altsem/gitu/pull/392/commits/39aa51df0d65569991eadee735bf4b4d596fab0a.